hack/jenkins/e2e.sh

```diff
@@ -1117,204 +1111,17 @@ export KUBE_SKIP_UPDATE=y
-            echo "Using published version $bucket/$build_version (from ${JENKINS_PUBLISHED_VERSION})"
-        fi
-        # At this point, we want to have the following vars set:
-        # - bucket
-        # - build_version
-        gsutil -m cp gs://kubernetes-release/${bucket}/${build_version}/kubernetes.tar.gz gs://kubernetes-release/${bucket}/${build_version}/kubernetes-test.tar.gz .
-
-        # Set by GKE-CI to change the CLUSTER_API_VERSION to the git version
-        if [[ ! -z ${E2E_SET_CLUSTER_API_VERSION:-} ]]; then
-            export CLUSTER_API_VERSION=$(echo ${build_version} | cut -c 2-)
-        fi
-    fi
-
-    if [[ ! "${CIRCLECI:-}" == "true" ]]; then
-        # Copy GCE keys so we don't keep cycling them.
-        # To set this up, you must know the <project>, <zone>, and <instance>
-        # on which your jenkins jobs are running. Then do:
-        #
-        # # SSH from your computer into the instance.
-        # $ gcloud compute ssh --project="<prj>" ssh --zone="<zone>" <instance>
-        #
-        # # Generate a key by ssh'ing from the instance into itself, then exit.
-        # $ gcloud compute ssh --project="<prj>" ssh --zone="<zone>" <instance>
-        # $ ^D
-        #
-        # # Copy the keys to the desired location (e.g. /var/lib/jenkins/gce_keys/).
-        # $ sudo mkdir -p /var/lib/jenkins/gce_keys/
-        # $ sudo cp ~/.ssh/google_compute_engine /var/lib/jenkins/gce_keys/
-        # $ sudo cp ~/.ssh/google_compute_engine.pub /var/lib/jenkins/gce_keys/
-        #
-        # # Move the permissions for the keys to Jenkins.
-        # $ sudo chown -R jenkins /var/lib/jenkins/gce_keys/
-        # $ sudo chgrp -R jenkins /var/lib/jenkins/gce_keys/
-        if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
-            echo "Skipping SSH key copying for AWS"
-        else
-            mkdir -p ${WORKSPACE}/.ssh/
-            cp /var/lib/jenkins/gce_keys/google_compute_engine ${WORKSPACE}/.ssh/
-            cp /var/lib/jenkins/gce_keys/google_compute_engine.pub ${WORKSPACE}/.ssh/
-        fi
-    fi
-
-    md5sum kubernetes*.tar.gz
-    tar -xzf kubernetes.tar.gz
-    tar -xzf kubernetes-test.tar.gz
-fi
-
-cd kubernetes
-
-# Have cmd/e2e run by goe2e.sh generate JUnit report in ${WORKSPACE}/junit*.xml
-ARTIFACTS=${WORKSPACE}/_artifacts
-mkdir -p ${ARTIFACTS}
-export E2E_REPORT_DIR=${ARTIFACTS}
-declare -r gcp_list_resources_script="./cluster/gce/list-resources.sh"
-declare -r gcp_resources_before="${ARTIFACTS}/gcp-resources-before.txt"
-declare -r gcp_resources_cluster_up="${ARTIFACTS}/gcp-resources-cluster-up.txt"
-declare -r gcp_resources_after="${ARTIFACTS}/gcp-resources-after.txt"
-# TODO(15492): figure out some way to run this script even if it doesn't exist
-# in the Kubernetes tarball.
-if [[ ( ${KUBERNETES_PROVIDER} == "gce" || ${KUBERNETES_PROVIDER} == "gke" ) && -x "${gcp_list_resources_script}" ]]; then
-  gcp_list_resources="true"
+if [[ -f ./hack/jenkins/e2e-runner.sh ]]; then
```

Comment:
My initial suggestion would be that the Jenkins jobs would always curl `hack/jenkins/e2e-runner.sh` from `master`, and then have `e2e-runner.sh` source the local `e2e.sh` for config, but this doesn't work for PR Jenkins.

The GitHub PR plugin sets `ghprbTargetBranch`, so maybe if it is set to `master` then source from the local checkout, otherwise download from GitHub?